platform.txt

```diff
@@ -53,7 +53,7 @@ compiler.libc.path={runtime.platform.path}/tools/sdk/libc/xtensa-lx106-elf
 compiler.cpreprocessor.flags=-D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I{compiler.sdk.path}/include" "-I{compiler.sdk.path}/{build.lwip_include}" "-I{compiler.libc.path}/include" "-I{build.path}/core"
 
 compiler.c.cmd=xtensa-lx106-elf-gcc
-compiler.c.flags=-c {compiler.warning_flags} -Os -g -Wpointer-arith -Wno-implicit-function-declaration -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -falign-functions=4 -MMD -std=gnu99 -ffunction-sections -fdata-sections {build.exception_flags} {build.sslflags}
+compiler.c.flags=-c {compiler.warning_flags} -std=c17 -Os -g -Wpointer-arith -Wno-implicit-function-declaration -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -falign-functions=4 -MMD -std=gnu99 -ffunction-sections -fdata-sections {build.exception_flags} {build.sslflags}
```

Comment:
Thanks, I believe that should now be fixed.  I can't test platform.io on my own (never was able to gett it running on either Linux or Windows), so I'll keep an eye on the CI system and see if something gets broken.

Reply:
as-is CI will fail, toolchain package version is still 4.8.2
tests/platformio.sh does some tricks to point to the Travis clone instead of trying to fetch things, so it needs a similiar thing that will point it to the toolchain, since it is hardcoded in the 'platform' files and platformIO's esp8266 'platform' is described outside of this repo.

I'd guess safe approach would be to
- get.py
- copy existing ~/.platformio/packages/toolchain-xtensa/package.json
- remove existing toolchain dir, replace with what get.py did get
- move package.json back
- pio ci, pio run should use it

Reply:
Wow, looks like I have to get PIO running in a VM to figure out how to automate that in the test script.  Will be a bit of work, but I'll give it a try.